waterbutler/providers/rushfiles/provider.py

```diff
@@ -187,9 +186,11 @@ async def intra_copy(self,
         ) as response:
             resp = await response.json()
             data = resp['Data']['ClientJournalEvent']['RfVirtualFile']
-        
-        dest_path.rename(data['PublicName'])
-        return RushFilesFileMetadata(data, dest_path), dest_path.identifier is None
+
+        dest_path.parts[-1]._id = data['InternalName']
+        move_src = dest_path
+        move_src.rename(data['PublicName'])        
+        return await self.intra_move(dest_provider, move_src, dest_path)
```

Comment:
First thing that `intra_move` is doing if `dest_path` has final id is deleting the file. If you set it to the `InternalName` of a cloned file, it will immediately be deleted. But not resetting it completely can also cause an error. If the `dest_path` existed, it's deleted on the line 172, and then could be deleted again in `move_intra`. The solution would be to either set id of the `dest_path` to None after deleting it on line 172, or remove it from `infra_copy` (leaving a comment with an explanation) and leave the task to `intra_move`. I'd suggest second option for the reasons I'll explain at the end.

Also with this code, `dest_path` and `move_src` have the same id.

```suggestion
        clone_result_path = dest_path.parent.child(data['PublicName'], _id=data['InternalName'])
        if clone_result_path == dest_path:
            # Cloned file is exactly the same as destination path. Can return right away.
            return RushFilesFileMetadata(data, clone_result_path ), True
        else:
            # Destination does not match (cloned file should be renamed or destination existed and we have a duplicate).
            # Leave renaming and/or deleting destination to intra_move.
          return self.intra_move(dest_provider, clone_result_path , dest_path)
```

With this code, we don't have to care at all whether the destination existed or not. So we can remove the deletion login (lines 171-172) from this method, and leave it to the intra_move.

Reply:
Thank you for your suggestion! I couldn't think of a better idea to avoid deletion.

Reply:
Maybe I haven't made myself clear. With this code, you can leave the deleting to the intra_copy, but only if you pass a valid state (of destination) in parameters. Right now, you're deleting the destination on line 171-172, but not marking it as not existing (not removing the id), and then passing to intra_move. In intra move, since the identifier of dest_path is not None, it will be deleted again causing error - you can't delete a file/folder twice.

You must either keep the state of valid: set id of dest_path to None after deleting it marking it as not existing; or not delete at all in intra_copy.